Dockerfile

```diff
@@ -2,26 +2,27 @@
 FROM oasislabs/rust
 
 RUN mkdir /service
-COPY ./ballot/service /service
+COPY ballot/service /service
 WORKDIR /service
 
 RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get -qq install \
     curl \
     python
-RUN curl --proto '=https' --tlsv1.2 -sSL https://get.oasis.dev | python \
-    /bin/bash -c  "source ~/.local/bin/oasis" >> ~/.bashrc \
-    oasis build
+RUN curl --proto '=https' --tlsv1.2 -sSL https://get.oasis.dev | python
+RUN export PATH=$PATH:$HOME/.local/bin \
+RUN oasis build
 
 # Build application
 FROM node:lts-alpine
 
-COPY ballot/app/package*.json /app
+RUN mkdir /app
+COPY ballot/app/package*.json /app/
 WORKDIR /app
 
 RUN npm install
 COPY ballot/app /app
-RUN mkdir -p /app/public/assets/bytecode
-COPY /service/target/service/ballot.wasm /app/public/assets/bytecode
+COPY ballot/service/target/service/ballot.wasm /app/public/assets
 
 RUN npm run build
-COPY /app/dist /app/static/staging
+RUN mkdir /app/static
+RUN mv /app/dist /app/static/staging
```

Comment:
[docker best practice is to merge `RUN` commands](https://stackoverflow.com/questions/39223249/multiple-run-vs-single-chained-run-in-dockerfile-what-is-better)